guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,61 @@
+# Motivation
+Our changelogs (reviewed by users/customers) are autogenerated from PR titles and git commit text.  The better this information is, the more we help our customers.  This information also helps:
+
+* reviewers understand changes
+* current+future team members understand the history of changes
+* patch managers understand risk/importance
+* when hunting for regressions, helps us better understand what is/isn’t likely to be a cause.
+
+# PR Metadata
+
+__PR title:__ Bug NNNN: A short sentence with capitalization and no period or the Jira tag for the issue “Feature TEAM-XXX:”
+
+* Try to describe the problem the user sees, rather than the code change you are doing
+  * I.e. “Console does not work when user clicks X” instead of “Fix drawer notification integration with underlying controller”
+* Prefix bug fixes with the bug number, “Bug XXXXX:” 
+  * (capitalize bug, space, number, then colon)
+* Do not let your PR title get truncated/wrapped into the description
+* Additional prefixes like what release the PR applies to are allowed, but they will be stripped from the changelog (the changelog for 4.4 doesn’t need to say “release-4.4” 
+  * Changelog strips anything between square brackets that has no spaces - i.e. [release-4.4] is stripped but [release 4.4] would not be
+    
+
+__PR description:__  Additional context reviewers need to understand your PR
+
+* Why was this change needed?
+* What changes are being made?  
+* Reference to any prior work(other PRs) or follow up work that will be needed.
+
+[Canonical PR text example](https://github.com/openshift/openshift-apiserver/pull/69)
+
+# Git Commit Best practices
+
+## Number of Commits
+
+Give careful consideration to the commits in your PR.  Multiple commits may make sense, but should each have a clear, logical, independent function.  Careful commit structure within a PR makes it easier to review the PR, easier to revert the PR if necessary, easier for someone doing “git blame” to follow the set of involved changes, and encourage smaller “chunks of work” (if it’s multiple commits, think about whether it should also be multiple PRs)
+
+Valid reasons to have separate commits:
+
+* Separating implementation work from generated code or vendor updates
+* Separating api work from implementation
+* Moving files around in a refactor (do the file move in one commit, edit the files in the second commit, so that the edits stand out in the commit diff)
+* Temporarily adding additional commits in response to reviewer feedback so reviewers can easily see how their feedback was addressed.  These commits should be squashed prior to merging.
+
+Invalid reasons to have separate commits
+* Separating tests+implementation
+* Committing+pushing your work each day/hour (use amend instead, or at least ensure you squash before merging).
+
+## Git Commit text
+
+__First line:__  An english language sentence with capitalization, no period, max 50 characters
```

Comment:
I would argue that area of modification is quite helpful. Here are some PR title examples from MCO:

- `cmd: Configure klog to match glog`: https://github.com/openshift/machine-config-operator/pull/1730
- `pkg/daemon: Add IgnitionVersion to Daemon`: https://github.com/openshift/machine-config-operator/pull/1729
- `mcs: Log User-Agent`: https://github.com/openshift/machine-config-operator/pull/1705

This is very helpful when going through history as well as when reviewing changes. Directly at a glance each PR's focus is quickly understood. On top of that, using this pattern helps folks avoid the all-the-things commits _unless_ it explicitly must be done (say -- refactoring variable names, etc.. though one could say the area would be `refactor:` :smile:) 

Please note, these examples do end up showing functionality rather than the problem a user sees. Combining "describe the problem" with "area of modification" would be quite a nice clarification overall!

Reply:
Are you suggesting this for the PR title or the commit text?  Given the commit text ends up in the release changelog i think it's undesirable in the commit text (first line).

having it in the PR title, or in the git text body seems ok, though it very much feels like a "ymmv" thing for how useful it is for different teams so i'm a little hesitant to put it in here as anything more than a "something to consider if you like it" suggestion.

Reply:
commit text (first line). I can accept your reasoning against it though @bparees. Generally I think the area pattern is helpful, but if things are flowing into the changelog and could be a bit awkward it would be understandable to note specify the area of change that way. It may be helpful in the text body, _but_ at that level I'd agree it's more of a "something to consider if you like it" at best.

Reply:
the area pattern is heavily used in installer if i recall.

/cc @abhinavdahiya should give his perspective as well.